src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqTouchstoneRepository.kt

```diff
@@ -171,36 +170,56 @@ class JooqTouchstoneRepository(
+                .from(TOUCHSTONE_COUNTRY)
+                .where(TOUCHSTONE_COUNTRY.TOUCHSTONE.eq(touchstoneId))
+    }
+
+    private fun touchstoneSources(touchstoneId: String): SelectConditionStep<Record2<Int, String>>
+    {
+        return dsl.select(DEMOGRAPHIC_SOURCE.ID, DEMOGRAPHIC_SOURCE.CODE.`as`("sourceCode"))
+                .from(DEMOGRAPHIC_SOURCE)
+                .join(TOUCHSTONE_DEMOGRAPHIC_SOURCE)
+                .on(DEMOGRAPHIC_SOURCE.ID.eq(TOUCHSTONE_DEMOGRAPHIC_SOURCE.DEMOGRAPHIC_SOURCE))
+                .where(TOUCHSTONE_DEMOGRAPHIC_SOURCE.TOUCHSTONE.eq(touchstoneId))
+    }
 
     private fun getDemographicStatisticTypesQuery(touchstoneId: String):
-            SelectOnConditionStep<Record6<Int, String, String, Boolean, String, String>>
+            SelectConditionStep<Record6<Int, String, String, Boolean, String, String>>
     {
-        val statsInTouchstoneCountriesAndSources =
-                dsl.selectDistinct(DEMOGRAPHIC_SOURCE.CODE.`as`("sourceCode"),
-                        DEMOGRAPHIC_STATISTIC.DEMOGRAPHIC_STATISTIC_TYPE.`as`("typeId"),
-                        TOUCHSTONE_COUNTRY.COUNTRY.`as`("country"))
-                        .from(DEMOGRAPHIC_STATISTIC)
-                        .join(TOUCHSTONE_COUNTRY)
-                        .on(DEMOGRAPHIC_STATISTIC.COUNTRY.eq(TOUCHSTONE_COUNTRY.COUNTRY))
-                        .join(DEMOGRAPHIC_SOURCE)
-                        .on(DEMOGRAPHIC_SOURCE.ID.eq(DEMOGRAPHIC_STATISTIC.DEMOGRAPHIC_SOURCE))
-                        .join(TOUCHSTONE_DEMOGRAPHIC_SOURCE)
-                        .on(DEMOGRAPHIC_SOURCE.ID.eq(TOUCHSTONE_DEMOGRAPHIC_SOURCE.DEMOGRAPHIC_SOURCE))
-                        .where(TOUCHSTONE_COUNTRY.TOUCHSTONE.eq(touchstoneId))
-                        .and(TOUCHSTONE_DEMOGRAPHIC_SOURCE.TOUCHSTONE.eq(touchstoneId))
-
-        return dsl.with("s")
-                .`as`(statsInTouchstoneCountriesAndSources)
+
+        val touchstoneSources = dsl.select(DEMOGRAPHIC_SOURCE.ID, DEMOGRAPHIC_SOURCE.CODE.`as`("sourceCode"))
+                .from(DEMOGRAPHIC_SOURCE)
+                .join(TOUCHSTONE_DEMOGRAPHIC_SOURCE)
+                .on(DEMOGRAPHIC_SOURCE.ID.eq(TOUCHSTONE_DEMOGRAPHIC_SOURCE.DEMOGRAPHIC_SOURCE))
+                .where(TOUCHSTONE_DEMOGRAPHIC_SOURCE.TOUCHSTONE.eq(touchstoneId))
+
+        val statsInTouchstoneSources = dsl.selectDistinct(DEMOGRAPHIC_STATISTIC.DEMOGRAPHIC_SOURCE,
+                DEMOGRAPHIC_STATISTIC.DEMOGRAPHIC_STATISTIC_TYPE.`as`("typeId"),
+                DEMOGRAPHIC_STATISTIC.COUNTRY.`as`("country"), field(name("s", "sourceCode"), String::class.java))
+                .from(DEMOGRAPHIC_STATISTIC)
+                .join(table(name("s")))
+                .on(DEMOGRAPHIC_STATISTIC.DEMOGRAPHIC_SOURCE.eq(field(name("s", "id"), Int::class.java)))
+
+        val countriesInTouchstone = countriesInTouchstone(touchstoneId)
+
+        return dsl.with("s").`as`(touchstoneSources)
```

Comment:
Maybe we could call the subquery `"sources"` rather than `"s"`? I know this isn't new in this branch, but reviewing it the second time I'm feeling less overwhelmed and being a bit more opinionated ;)

Similarly, maybe `"sts"` could become `"statsInSources"`?

Finally, to make sure these don't get out of sync, would it be worth pulling these strings into consts?

e.g. 
```
val TOUCHSTONE_SOURCES = "touchstoneSources"
val STATS_IN_SOURCES = "statsInTouchstoneSources"
...
field(name(STATS_IN_SOURCES, "sourceCode")
...
etc.
```